tools/yamcs/proves_adapter.py

```
@@ -50,118 +62,257 @@ def _crc16_ccitt(data: bytes) -> int:
+        "vc_count": frame[3],
+        "length": len(frame),
+        "frame_hex": frame.hex(),
+    }
+    try:
+        with log_path.open("a") as stream:
+            stream.write(json.dumps(record) + "\n")
+    except OSError as exc:
+        print(f"[TM] failed to write {log_path}: {exc}")
+
+
 def _forward_tm_serial(
```

Ruff (0.15.20)

[warning] Too many statements (66 > 50)

(PLR0915)